src/pytorch_lightning/strategies/tpu_spawn.py

```diff
@@ -212,9 +212,9 @@ def reduce(
 
         return output
 
-    def _worker_setup(self, process_idx: int) -> None:
+    def setup_distributed(self) -> None:
```

Comment:
Should this be `setup_environment`?

This implementation matches exactly `XLAStrategy.setup_environment`: https://github.com/Lightning-AI/lightning/blob/7ff297bdffbc53beb6ffc999c013341d3a0d09c8/src/lightning_lite/strategies/xla.py#L97-L100

Right now, this difference between strategies is causing https://github.com/Lightning-AI/lightning/pull/14926#discussion_r983009669

Alternatively, if we want `setup_distributed` here then `XLAStrateg`y should change `setup_environment -> setup_distributed` too.

Reply:
`setup_environment` calls `setup_distributed` and does nothing else except calling `super().setup_environment`. 

This may look weird to you, why not collapse the two similar methods into one? 

The answer is very simple: It wasn't always like this. For historical reasons, these two methods had to handle different things, and they existed also to be overridden differently in the subclasses, e.g. TPUSpawn. Overtime, we have simplified things, but it is a slow process and the strategies are just in a constant state of incompleteness. 

In summary, this PR makes 

`setup_environment` in DDP and DDPSpawn identical, because they should be!
`setup_distributed` in DDP and DDPSpawn identical, because they should be!


 Maybe this helps in https://github.com/Lightning-AI/lightning/pull/14926#discussion_r983009669, but I'm sure the accelerator issue is unrelated. I will take a look there.

Reply:
Resolving, we did

> Alternatively, if we want setup_distributed here then XLAStrategy should change setup_environment -> setup_distributed too.